SYCL/DeviceLib/sycl_ext_intel_math_fp16.cpp

```diff
@@ -0,0 +1,72 @@
+// RUN: %clangxx -fsycl %s -o %t.out
+// RUN: %CPU_RUN_PLACEHOLDER %t.out
+// RUN: %ACC_RUN_PLACEHOLDER %t.out
+// RUN: %GPU_RUN_PLACEHOLDER %t.out
+
+// RUN: %clangxx -fsycl -fno-builtin -fsycl-device-lib-jit-link %s -o %t.out
+// RUN: %CPU_RUN_PLACEHOLDER %t.out
+// RUN: %ACC_RUN_PLACEHOLDER %t.out
+// RUN: %GPU_RUN_PLACEHOLDER %t.out
+//
+// UNSUPPORTED: cuda || hip
+
+#include "imf_utils.hpp"
+#include <sycl/ext/intel/math.hpp>
+
+class is_close {
+public:
+  bool operator()(sycl::half x, sycl::half y) {
+    if (sycl::fabs((x - y) / (x + y)) <
+        std::numeric_limits<sycl::half>::epsilon() * 2)
+      return true;
+    return false;
```

Comment:
Nit; I would prefer something like
```suggestion
    return sycl::fabs((x - y) / (x + y)) <
        std::numeric_limits<sycl::half>::epsilon() * 2;
```
and similar for the other check operators.

Reply:
This comment was only applied to one function, but it also applies to `is_close2` and `is_close` in the other tests.

Reply:
Done.
Thanks very much.